ports/stm32/subghz_radio.c

```diff
@@ -0,0 +1,193 @@
+    enum { ARG_handler, ARG_trigger };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_handler, MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_trigger, MP_ARG_INT, {.u_int = GPIO_MODE_IT_RISING} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    if (n_args > 1 || kw_args->used != 0) {
+        if (args[ARG_trigger].u_int != GPIO_MODE_IT_RISING) {
+            // trigger argument is accepted for compatibility with Pin.irq, but has to be RISING
+            mp_raise_ValueError(MP_ERROR_TEXT("unsupported trigger"));
+        }
+
+        mp_obj_t new_cb = args[ARG_handler].u_obj;
+        MP_STATE_PORT(machine_subghz_radio_callback) = new_cb;
+    }
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(machine_subghz_radio_irq_obj, 1, machine_subghz_radio_irq);
+
+STATIC mp_obj_t machine_subghz_radio_irq_en(mp_obj_t self_in, mp_obj_t enable) {
+    if (mp_obj_is_true(enable)) {
+        // Because this interrupt is level triggered but the IRQ can't reset it (see above),
+        // it will remain asserted (and set pending) until cleared by Python code
+        HAL_NVIC_ClearPendingIRQ(SUBGHZ_Radio_IRQn);
+        HAL_NVIC_EnableIRQ(SUBGHZ_Radio_IRQn);
+    } else {
+        HAL_NVIC_DisableIRQ(SUBGHZ_Radio_IRQn);
+    }
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(machine_subghz_radio_irq_en_obj, machine_subghz_radio_irq_en);
+
+STATIC mp_obj_t machine_subghz_radio_is_busy(mp_obj_t self_in) {
+    (void)self_in;
+
+    // Read the raw unmasked busy signal. This should be checked before driving
+    // CS low to start a command.
+    //
+    // Reads the raw RFBUSYS not the masked RFBUSYMS, in contradiction to EM0453
+    // 6.3 "Radio busy management". This is because the RFBUSYMS signal doesn't
+    // seem to match the reference manual. Observed behaviour matches this bug
+    // report instead: https://community.st.com/s/question/0D53W000014zFx9SAE
+    //
+    // Reading RFBUSYS won't cause any problems here provided a new SPI command
+    // isn't immediately after the previous command, which shouldn't be possible
+    // with MicroPython.
+    return mp_obj_new_bool(LL_PWR_IsActiveFlag_RFBUSYS());
+
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_subghz_radio_is_busy_obj, machine_subghz_radio_is_busy);
+
+
+STATIC const mp_rom_map_elem_t mp_subghz_radio_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_spi), MP_ROM_PTR(&machine_subghz_radio_spi_obj) },
+    { MP_ROM_QSTR(MP_QSTR_cs), MP_ROM_PTR(&machine_subghz_radio_cs_obj) },
+    { MP_ROM_QSTR(MP_QSTR_irq), MP_ROM_PTR(&machine_subghz_radio_irq_obj) },
+    { MP_ROM_QSTR(MP_QSTR_irq_en), MP_ROM_PTR(&machine_subghz_radio_irq_en_obj) },
+    { MP_ROM_QSTR(MP_QSTR_is_busy), MP_ROM_PTR(&machine_subghz_radio_is_busy_obj) },
```

Comment:
Make these into 3 functions (subghz_cs, subghz_irq, subghz_is_busy) in the `stm` module.

Reply:
Done. Needed a fourth function as well, to enable/disable the HSE32 oscillator.